python/test/integration/test_mean.py

```diff
@@ -0,0 +1,32 @@
+import unittest
+
+from opendp.mean import *
+
+enable_features("floating-point")
+
+
+class DPMeanTest(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.data = '59,1,9,1,0,1\n31,0,1,3,17000,0\n36,1,11,1,0,1\n54,1,11,1,9100,1\n39,0,5,3,37000,0\n34,0,9,1,0,1\n'\
+                    '93,1,8,1,6000,1\n69,0,13,1,350000,1\n40,1,11,3,33000,1\n27,1,11,1,25000,0\n59,1,13,1,49000,1\n' \
+                    '31,1,11,3,0,1\n73,1,13,4,35500,0\n89,1,9,1,4000,1\n39,1,10,3,15000,0\n51,1,13,4,120000,1\n' \
+                    '32,0,9,1,13000,0\n52,0,11,2,45000,0\n24,0,7,1,0,0\n48,1,10,1,4300,1\n51,0,1,3,16000,1\n' \
+                    '43,1,14,1,365000,1\n29,0,4,3,20000,0\n44,1,15,1,17900,1\n87,1,8,1,3600,0\n27,1,11,3,10800,0\n' \
+                    '58,0,13,1,60900,1\n32,1,11,3,25000,1\n'
+
+    def test_dp_mean_compute(self):
+        """
+        A test to ensure that the computation chain for DP Mean completes and returns a value
+        :return:
+        """
+        epsilon = 1.
+        col_names = ["A", "B", "C", "D", "E"]
+        column = "E"
+        dp_mean = DPMean(self.data)
```

Comment:
This seems backwards to me. Typically computation parameters come first, and the data may or may not be used.

PSI had a class interface similar to what you're writing. The class init would accept computation parameters, and then you could use different methods on the class instance to compute accuracies or make a release.

If you do want to keep the class interface, you could make a base class that implements the shift chainer and __call__, and then inherit from that in DPMean. In DPMean's __init__, conduct the binary search and save the discovered chain as a class attribute. Then it would behave similarly to other measurements and transformations, but you could add an additional method for accuracy estimation.

Reply:
The only gain I see at this point from the class interface is that you can glue additional helper methods onto it like accuracy. Not sure if that's worth adopting classes over just plain simple functions that wrap multiple constructors up in one.